src/libcore/tests/slice.rs

```diff
@@ -134,6 +134,26 @@ fn test_chunks_nth() {
     assert_eq!(c2.next(), None);
 }
 
+#[test]
+fn test_chunks_nth_back() {
+    let v: &[i32] = &[0, 1, 2, 3, 4, 5];
+    let mut c = v.chunks(2);
+    assert_eq!(c.nth_back(1).unwrap(), &[2, 3]);
+    assert_eq!(c.next().unwrap(), &[0, 1]);
+    assert_eq!(c.next(), None);
+
+    let v2: &[i32] = &[0, 1, 2, 3, 4];
+    let mut c2 = v2.chunks(3);
+    assert_eq!(c2.nth_back(1).unwrap(), &[0, 1]);
```

Comment:
This assertion does not currently pass as of 1.36.0-nightly (a3404557c 2019-05-03). Is there an intentional behavior change in this PR?

Reply:
I did not mean to intentionally change behavior with this PR. However, I would try to understand why this should be happening and update.

Reply:
`c2.nth_back(1).unwrap()` should return `&[0, 1, 2]` here, right? Since that's the first of two chunks, so the second chunk from the back.

Reply:
As per the documentation for Chunks: [https://github.com/rust-lang/rust/blob/master/src/libcore/slice/mod.rs#L4052-L4056](url), chunks are non-overlapping, and thereby a chunk of chunk_size 3, should have [ {2,3,4}, {0,1} ] as the slices, for nth_back. Do let me know if I misunderstood anything.

Reply:
`nth_back` is just a method on `DoubleEndedIterator`, it doesn't actually change the items. `[0, 1, 2, 3, 4].chunks(3)` returns an iterator over the items `&[0, 1, 2]` and `&[3, 4]`, in that order, so calling `.nth_back(1)` on it should return `&[0, 1, 2]` (the second element from the back).

Note that your description matches the behavior of `rchunks`, which does have its first chunk at the end of the slice.

Reply:
Hi @timvermeulen I have changed the implementation and pushed the changes... please do let me know if you have any more review comments or any regression fails..

Reply:
Looks like something went wrong, the diff is huge now...

Reply:
ya this happened i believe because of rebasing to changes in master... shall i post the patch for the changes??

Reply:
I'm not really an expert when it comes to git, sorry.

As for your changes to `nth` – I think it's a lot more complicated than necessary. For starters, you don't need `checked_rem` because `self.chunk_size` is guaranteed to be non-zero, and you could replace your usage of `checked_sub` with `saturating_sub`.

For inspiration, you could have a looks at my `RChunks::nth` implementation [here](https://github.com/rust-lang/rust/pull/60555), which should probably be pretty similar.